File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        map<int, int> dict;

        for(auto i : nums){
            if(dict.find(i) != dict.end()) {
                return true;
            }

            dict[i] = i;
        }

        return false;
    }
};

File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length <= 1)
            return nums.length;

        int i = 1;
        int j = 1;

        while (i < nums.length) {
            if (nums[i] != nums[j - 1]) {
                nums[j++] = nums[i];
            }
            i++;
        }

        return j;
    }
}